src/s3/activation.ts

```diff
@@ -10,26 +10,33 @@ import { createFolderCommand } from './commands/createFolder'
 import { deleteBucketCommand } from './commands/deleteBucket'
 import { deleteFileCommand } from './commands/deleteFile'
 import { downloadFileAsCommand } from './commands/downloadFileAs'
+import { openFileCommand } from './commands/openFile'
 import { uploadFileCommand } from './commands/uploadFile'
 import { uploadFileToParentCommand } from './commands/uploadFileToParent'
 import { S3BucketNode } from './explorer/s3BucketNode'
 import { S3FolderNode } from './explorer/s3FolderNode'
 import { S3Node } from './explorer/s3Nodes'
 import { S3FileNode } from './explorer/s3FileNode'
 import { ext } from '../shared/extensionGlobals'
+import { S3FileViewerManager } from './util/fileViewerManager'
 import { ExtContext } from '../shared/extensions'
 
 /**
  * Activates S3 components.
  */
+
 export async function activate(ctx: ExtContext): Promise<void> {
+    ext.fileViewerManager = new S3FileViewerManager()
```

Comment:
this should be done in extension.ts where all other `ext` globals are assigned. spreading out the init across the project leads to confusion, harder to reason about, and harder to see the order of operations.